### PR TITLE
fix(addons): Use keywords exclusively — fix Meson preset bypass, add edge-case tests, and update docs

### LINKED ISSUE #20528: Update POT file (xgettext) plugin: option to use keywords exclusively
### Describe the problem

I work on an old Java project where unfortunately we use some of the xgettext keywords for non translation related string creation. I solve this in my script by adding a "--keyword" without parameter. According to xgettext manual this deletes all default keywords. I add another "--keyword tr" for my translation method(tr). 

### Solution brainstorm

My idea would be e.g. a checkbox to state that only the keywords defined in the additional keyword field  should be used and not the default ones.

### Describe alternatives you have considered

_No response_

### Screenshots

_No response_

### Additional context

from xgettext manual:
`--keyword[=keywordspec]´
    Additional keyword to be looked for (**without keywordspec means not to use default keywords**). If keywordspec is a C identifer id, xgettext looks for strings in the first argument of each call to the function or macro id. If keywordspec is of the form `id:argnum´, xgettext looks for strings in the argnumth argument of the call. If keywordspec is of the form `id:argnum1,argnum2´, xgettext looks for strings in the argnum1st argument and in the argnum2nd argument of the call, and treats them as singular/plural variants for a message with plural handling.
    The default keyword specifications, which are always looked for if not explicitly disabled, are gettext, dgettext:2, dcgettext:2, ngettext:1,2, dngettext:2,3, dcngettext:2,3, and gettext_noop.
    This option has an effect with most languages, namely C, C++, ObjectiveC, Shell, Python, Lisp, EmacsLisp, librep, Java, C#, awk, Tcl, Perl, PHP, GCC-source, Glade.

### PR DESCRIPTION
**Pull request by Kakarot30**
## Description
This PR implements the **Use keywords exclusively** checkbox for the `weblate.gettext.xgettext` and `weblate.gettext.meson` add-ons. When enabled, Weblate passes a bare `--keyword` to xgettext immediately before the user-defined `--keyword=<name>`. Per the xgettext manual, a bare `--keyword` disables all built-in default keywords so that only the user's custom-defined keywords are recognized.
### Key Problems Solved
1. **Meson Preset Keywords Bypass**:
   `MesonAddon.get_extra_xgettext_args()` previously prepended preset GLib keywords (e.g. `--keyword=_`, `--keyword=N_`) before the parametrized xgettext arguments. Because a bare `--keyword` only disables built-in defaults (and not custom arguments already specified earlier on the command line), Meson users selecting "Use keywords exclusively" still extracted GLib call patterns.
   * **Fix**: When `keyword_exclusive` is enabled, all `--keyword=...` entries from the selected preset are stripped out entirely, ensuring only the custom keyword is processed.
2. **Add-on Configuration Snippets**:
   The `keyword_exclusive` field was missing from the autogenerated parameter tables in `docs/snippets/addons-autogenerated.rst` for both add-ons. 
   * **Fix**: Regenerated/inserted the missing `keyword_exclusive` rows into the respective configuration tables.
3. **Pre-commit Cleanups**:
   * Removed backslash-escaped quotes (`\"`) in `tests.py` that caused syntax errors on Ruff checks.
   * Replaced non-ASCII ellipsis characters (`…` / U+2026) in python docstrings and comments with ASCII `...` to prevent `doccmd` UnicodeDecodeError crashes.
   * Suppressed Ruff `unused-class-method-argument` warnings in `tests.py` for test mock signature compliance using `# noqa: ARG002`.
---

Closes #20528

### CODE REVIEW

**Comment by nijel** [comment]
I think that two commits should be removed from this:

* https://github.com/WeblateOrg/weblate/pull/20556/changes/9f3efaa41585fbc1e17440f31c2ac726cc6b7847 just changes override types, should not be needed.
* https://github.com/WeblateOrg/weblate/pull/20556/changes/326337aef44db152ac2e0a1bbd27043d6b4d1cd3 is changing existing documentation and there should be no issue in using Unicode there.

File: weblate/addons/tests.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 import os
 import re
 import shutil
-import subprocess  # ruff: ignore[suspicious-subprocess-import]
+import subprocess  # noqa: S404
```
Comment (nijel) [inline weblate/addons/tests.py]: Please avoid changing existing suppressions. The human-readable identifiers are way easier to understand than the old error codes:

```suggestion
import subprocess  # ruff: ignore[suspicious-subprocess-import]
```

**Comment by nijel** [comment]
@codex review this

**Comment by nijel** [comment]
Do you know you can force push to a branch and update the pull request by that? There is no need to close the old pull request and open a new one just to change the content.

**Comment by nijel** [comment]
@codex review this

**Comment by Kakarot30** [comment]
Yes , Sorry for making so many prs i was just focusing on how to pass that pre commit check because everything else was looking good

**Comment by nijel** [comment]
The pre-commit.ci is broken. They fetched a broken module and apparently keep it cached forever, so you can ignore this one.

File: weblate/addons/tests.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
...
```
Comment (nijel) [inline weblate/addons/tests.py]:
```suggestion
import subprocess  # ruff: ignore[suspicious-subprocess-import]
```

**Comment by nijel** [comment]
@codex review

**Comment by nijel** [comment]
Merged, thanks for your contribution!
